test/unit/spec/VerifyCustomFactor_spec.js

```diff
@@ -0,0 +1,282 @@
+      {
+        controller: controller,
+      },
+      {
+        name: 'AuthApiError',
+        message: message,
+        statusCode: code,
+        xhr: resError,
+      },
+    ]);
+  }
+
+  function expectSubtitleToBe(test, desiredSubtitle) {
+    expect(test.form.subtitleText()).toBe(desiredSubtitle);
+  }
+
+  async function setup(res, selectedFactorProps, settings, lastFailedChallengeFactorData, languagesResponse , startRouter) {
+    const setNextResponse = Util.mockAjax();
+    const baseUrl = 'https://foo.com';
+    const authClient = getAuthClient({
+      authParams: {
+        issuer: baseUrl,
+        transformErrorXHR: LoginUtil.transformErrorXHR
+      }
+    });
+    const successSpy = jest.fn();
+    const afterErrorHandler =  jest.fn();
+    const router = createRouter(baseUrl, authClient, successSpy, settings, startRouter);
+
+    router.on('afterError', afterErrorHandler);
+    if (startRouter) {
+      if (typeof startRouter !== 'boolean') {
+        throw new Error('startRouter parameter should be boolean');
+      }
+      await Expect.waitForPrimaryAuth();
+    }
+
+    setNextResponse(res);
+    if (languagesResponse) {
+      setNextResponse(languagesResponse);
+    }
+    router.refreshAuthState('dummy-token');
+    await Expect.waitForMfaVerify();
+
+    router.appState.set('lastFailedChallengeFactorData', lastFailedChallengeFactorData);
+
+    if (selectedFactorProps) {
+      const factors = router.appState.get('factors');
+      const selectedFactor = factors.findWhere(selectedFactorProps);
+      const provider = selectedFactor.get('provider');
+      const factorType = selectedFactor.get('factorType');
+
+      if (provider === 'CUSTOM' && factorType === 'claims_provider') {
+        setNextResponse(resChallengeClaimsProvider);
+        router.verifyClaimsFactor();
+        await Expect.waitForVerifyCustomFactor();
+      }
+    }
+
+    const $forms = $sandbox.find('.o-form');
+
+    let forms = _.map($forms, function(form) {
+      return new VerifyCustomFactorForm($(form));
+    });
+
+    if (forms.length === 1) {
+      forms = forms[0];
+    }
+    const beacon = new Beacon($sandbox);
+
+    return {
+      router: router,
+      form: forms,
+      beacon: beacon,
+      ac: authClient,
+      setNextResponse: setNextResponse,
+      successSpy: successSpy,
+      afterErrorHandler: afterErrorHandler,
+    };
+  }
+
+  const setupClaimsProviderFactorWithIntrospect = _.partial(setup, resAllFactors, {
+    factorType: 'claims_provider',
+    provider: 'CUSTOM',
+  });
+
+  Expect.describe('Claims Provider Factor', function() {
+    Expect.describe('when SKIP_IDP_FACTOR_VERIFICATION_BUTTON FF is on', function() {
+      const lastFailedChallengeFactorData = {
+        errorMessage: 'Verify failed.'
+      };
+
+      let settings = {};
+
+      beforeEach(() => {
+        settings = { 'features.skipIdpFactorVerificationBtn': true };
+      });
+
+      itp('hides the button bar when', function() {
```

Comment:
just "hides the button bar". There is an extra "when" at the end of the sentence that we can remove